lib/admin/services/process_file.rb

```diff
@@ -5,60 +5,46 @@ module Services
     # necessary, of an XLSX file.
     class ProcessFile
 
-      def initialize(xlsx_file,
-                     file_validator: Validators::FileValidator,
-                     marshaler: Processing::Marshaler,
-                     xl_parser: RubyXL::Parser)
-        @xlsx_file = xlsx_file
-        @file_validator = file_validator
-        @marshaler = marshaler
-        @xl_parser = xl_parser
+      attr_reader :errors
+
+      def initialize(xlsx_path)
+        @xlsx_path = xlsx_path
+        @errors = []
       end
 
       def call
-        raise 'No file specified' unless @xlsx_file
-        rubyxl_workbook = @xl_parser.parse(file.path)
-        as_hashes = Parsers::Workbook.new(rubyxl_workbook).call
-        file_validations = @file_validator.new(as_hashes)
+        raise 'No file specified' unless @xlsx_path
+        rubyxl_workbook = RubyXL::Parser.parse(@xlsx_path)
+        mediators_as_hashes, @blacklist = Parsers::Workbook.new(rubyxl_workbook).call
```

Comment:
This is smelly. Requires deep knowledge that the `call` method returns an array in a particular order. Could `mediators_as_hashes` and `blacklist` be instance methods on `Parsers::Workbook`?